backend/app/datasets/query_stats_daily/service.py

```diff
@@ -74,3 +158,236 @@ def delete_query_stats_daily(
             .delete(synchronize_session=False)
         )
         self.db.commit()
+
+    @staticmethod
+    def _sort_stats_by_date_and_consumer(
+        stats: list[DatasetQueryStatsDailyResponse],
+    ) -> list[DatasetQueryStatsDailyResponse]:
+        """Sort stats by date, then consumer name, then consumer ID."""
+        return sorted(
+            stats,
+            key=lambda stat: (
+                stat.date,
+                stat.consumer_data_product_name or EMPTY_STRING_FALLBACK,
+                str(stat.consumer_data_product_id),
+            ),
+        )
+
+    def _group_low_volume_consumers(
```

Comment:
This function is a bit of a duplicate of what the front end is doing to generate to one of the charts? Maybe we can just include it in the end result to only have to calculate this once?